modAionImpl/src/org/aion/zero/impl/sync/SyncStats.java

```diff
@@ -151,32 +160,23 @@ public String dumpRequestStats() {
     }
 
     /**
-     * Returns a log stream containing a list of peers ordered by the total number of blocks
-     * received from each peer used to determine who is providing the majority of blocks, i.e. top
-     * seeds.
+     * Updates the total number of blocks received/imported/stored from each seed peer
      *
-     * @return log stream with peers statistical data on seeds
+     * @param nodeId peer node display Id
+     * @param blocks total number of blocks
+     * @param type type of the block: received, imported, stored
      */
-    public String dumpTopSeedsStats() {
+    public void updatePeerBlocks(String nodeId, int blocks, BlockType type) {
         if (seedsEnabled) {
-            return this.seedsTracker.dumpTopSeedsStats();
-        } else {
-            return "";
+            seedsTracker.updatePeerBlocksByType(nodeId, blocks, type);
         }
     }
 
     /**
-     * Updates the total number of blocks received from each seed peer
+     * Obtains a map of seed peers ordered by the total number of imported blocks
      *
-     * @param nodeId peer node display Id
-     * @param receivedBlocks total number of blocks received
+     * @return map of total blocks receivedby peer and sorted in descending order
      */
-    public void updatePeerReceivedBlocks(String nodeId, int receivedBlocks) {
-        if (seedsEnabled) {
-            this.seedsTracker.updatePeerReceivedBlocks(nodeId, receivedBlocks);
-        }
-    }
-
     @VisibleForTesting
     Map<String, Integer> getReceivedBlocksByPeer() {
```

Comment:
`getReceivedBlocksByPeer`, `getImportedBlocksByPeer`, `getStoredBlocksByPeer` need to also be refactored into a single method using the enum as parameter